pkg/utils/instance/instance.go

```diff
@@ -1,72 +1,50 @@
 package instance
 
 import (
-	"strings"
-
 	"github.com/aws/aws-sdk-go-v2/aws"
 	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
 )
 
 // IsARMInstanceType returns true if the instance type is ARM architecture
 func IsARMInstanceType(instanceType string) bool {
-	return strings.HasPrefix(instanceType, "a1") ||
-		strings.HasPrefix(instanceType, "t4g") ||
-		strings.HasPrefix(instanceType, "m6g") ||
-		strings.HasPrefix(instanceType, "m7g") ||
-		strings.HasPrefix(instanceType, "m8g") ||
-		strings.HasPrefix(instanceType, "r8g") ||
-		strings.HasPrefix(instanceType, "c8g") ||
-		strings.HasPrefix(instanceType, "c6g") ||
-		strings.HasPrefix(instanceType, "c7g") ||
-		strings.HasPrefix(instanceType, "r6g") ||
-		strings.HasPrefix(instanceType, "r7g") ||
-		strings.HasPrefix(instanceType, "im4g") ||
-		strings.HasPrefix(instanceType, "is4g") ||
-		strings.HasPrefix(instanceType, "g5g") ||
-		strings.HasPrefix(instanceType, "hpc7g") ||
-		strings.HasPrefix(instanceType, "x2g")
+	if InstanceTypesMap[instanceType].CPUArch == "arm64" {
+		return true
+	}
+	return false
 }
 
 // IsGPUInstanceType returns true if the instance type is GPU optimised
 func IsGPUInstanceType(instanceType string) bool {
-	return IsNvidiaInstanceType(instanceType) ||
-		IsInferentiaInstanceType(instanceType) ||
-		IsTrainiumInstanceType(instanceType)
+	itype := InstanceTypesMap[instanceType]
+	return itype.NvidiaGPUSupported || itype.NeuronSupported
 }
 
 // IsNeuronInstanceType returns true if the instance type requires AWS Neuron
 func IsNeuronInstanceType(instanceType string) bool {
-	return IsInferentiaInstanceType(instanceType) ||
-		IsTrainiumInstanceType(instanceType)
+	return InstanceTypesMap[instanceType].NeuronSupported
 }
 
 // IsARMGPUInstanceType returns true if the instance type is ARM-GPU architecture
 func IsARMGPUInstanceType(instanceType string) bool {
-	return strings.HasPrefix(instanceType, "g5g")
+	itype := InstanceTypesMap[instanceType]
+	return itype.CPUArch == "arm64" && itype.NvidiaGPUSupported
 }
 
 // IsNvidiaInstanceType returns true if the instance type has NVIDIA accelerated hardware
 func IsNvidiaInstanceType(instanceType string) bool {
-	return strings.HasPrefix(instanceType, "p2") ||
-		strings.HasPrefix(instanceType, "p3") ||
-		strings.HasPrefix(instanceType, "p4") ||
-		strings.HasPrefix(instanceType, "p5") ||
-		strings.HasPrefix(instanceType, "g3") ||
-		strings.HasPrefix(instanceType, "g4") ||
-		strings.HasPrefix(instanceType, "g5") ||
-		strings.HasPrefix(instanceType, "g6")
+	return InstanceTypesMap[instanceType].NvidiaGPUSupported
 }
 
 // IsInferentiaInstanceType returns true if the instance type requires AWS Neuron
 func IsInferentiaInstanceType(instanceType string) bool {
```

Comment:
we shouldn't need separate checks for Inferentia vs Trainium - it should just be a check for `NeuronSupported`, similar to `NvidiaGPUSupported`


We could proceed with this current change but we'll have to exhaustively check the Neuron device types (`Inferentia`, Inferentia2`, etc. Same for `Trainium`, `Trainium2`) or do a prefix/startswith check instead

Reply:
added explicit checks for Inferentia/Inferentia2 and Trainium/Trainium2.

Reply:
we can revisit the places we are making the calls and see if we want to consolidate. for this PR i'd like to get the checks right (as-is! but killing things like "g3.xlarge" "p2.large" that are no longer there)

Reply:
agreed - I was about to comment about p2 but thats an issue we can tackle separately; this just helps to remove the hardcoding of instance type name prefixes which is awesome

I think once the checks are green, we are good to merge

Reply:
thanks @bryantbiggs